localstack-core/localstack/services/stepfunctions/asl/component/state/state_execution/state_map/iteration/distributed_iteration_component.py

```diff
@@ -150,25 +152,24 @@ def _eval_body(self, env: Environment) -> None:
         )
         env.map_run_record_pool_manager.add(self._map_run_record)
 
-        env.event_history.add_event(
+        env.execution_event_manager.add_event(
             context=env.event_history_context,
-            hist_type_event=HistoryEventType.MapRunStarted,
-            event_detail=EventDetails(
+            event_type=HistoryEventType.MapRunStarted,
+            event_details=EventDetails(
                 mapRunStartedEventDetails=MapRunStartedEventDetails(
                     mapRunArn=self._map_run_record.map_run_arn
                 )
             ),
         )
 
-        execution_event_history = env.event_history
+        execution_event_history = env.execution_event_manager
         try:
             if self._eval_input.item_reader:
                 self._eval_input.item_reader.eval(env=env)
             else:
                 env.stack.append(self._eval_input.input_items)
 
-            # TODO: investigate if this is truly propagated also to eventual sub programs in map run states.
-            env.event_history = EventHistory()
+            env.execution_event_manager = ExecutionEventManager()
```

Comment:
Indeed, the initial intention was to make sure this object would be associated solely with the recording of [HistoryEvent](https://docs.aws.amazon.com/step-functions/latest/apireference/API_HistoryEvent.html) objects. This is as the object also handle the logging of logs.
I think I would prefer to remove the `execution_` prefix; `EventManager` would position this object as the entry point for execution events i/o, rather than only history events objects. Unfortunately I also don't see a more specific alternative to "manager".

Reply:
I don't see a big problem with a smart event history that also handles the logging concern, but I also understand the coordinator role of the event manager. Not overly important, it just caught my attention (no strong opinion).